src/services/projects.js

```diff
@@ -555,46 +654,84 @@ module.exports = class ProjectsHelper {
 						let checkRegex = utils.checkRegexPattern(taskEntityType, fieldData)
 						if (checkRegex) {
 							throw {
-								message: `Special characters not allowed in ${taskEntityType.value}`,
-								error: utils.errorObject(common.TASKS, taskEntityType.value),
+								statusCode: httpStatusCode.bad_request,
+								responseCode: 'CLIENT_ERROR',
+								error: utils.errorObject(
+									common.TASKS,
+									taskEntityType.value,
+									`Special characters not allowed in ${taskEntityType.value}`
+								),
 							}
 						}
 					}
-				})
+				}
 				// TODO: Get file types from products teams and add validation for them
 				if (task.allow_evidences == common.TRUE && task.evidence_details.file_types.length < 1) {
 					throw {
-						message: 'FILE_TYPE_NOT_SELECTED',
-						error: 'utils.errorObject(common.BODY, common.FILE_TYPE)',
+						error: utils.errorObject(common.BODY, common.FILE_TYPE, 'File type not selected'),
+					}
+				} else if (task.allow_evidences == common.TRUE && task.evidence_details.file_types.length > 0) {
+					let difference = _.difference(task.evidence_details.file_types, allowed_fileTypes_entities)
+
+					if (difference.length > 0) {
+						throw {
+							error: utils.errorObject(common.TASK_EVIDENCE, common.FILE_TYPE, 'Invalid file type'),
+						}
 					}
 				}
 
 				if (task.learning_resources && task.learning_resources.length > 0) {
-					let subTaskEntityTypes = await entityModelMappingQuery.findEntityTypesAndEntities(
-						{
-							model: common.SUBTASKS,
-							status: common.STATUS_ACTIVE,
-						},
-						userDetails.organization_id,
-						['value', 'validations']
-					)
 					task.learning_resources.forEach((learningResource) => {
-						let validateURL = utils.checkRegexPattern(subTaskEntityTypes[0], learningResource.url)
+						let validateName = utils.checkRegexPattern(
+							subTaskEntityTypesMapping['name'],
+							learningResource.name
+						)
+						if (validateName) {
+							throw {
+								error: utils.errorObject(
+									common.BODY,
+									common.LEARNING_RESOURCE,
```

Comment:
@adithyadinesh0412 learning resource of task right check with frontend what will be the params they are expecting

Reply:
Sure , will check with frontend team